packages/emitter-framework/src/python/components/class-declaration/class-method.tsx

```diff
@@ -0,0 +1,124 @@
+import { type Children, createContext, List, splitProps, useContext } from "@alloy-js/core";
+import * as py from "@alloy-js/python";
+import type { Operation } from "@typespec/compiler";
+import type { Typekit } from "@typespec/compiler/typekit";
+import { useTsp } from "../../../core/index.js";
+import { buildParameterDescriptors, getReturnType } from "../../utils/operation.js";
+import { TypeExpression } from "../type-expression/type-expression.jsx";
+
+export const MethodContext = createContext<"method" | "static" | "class" | undefined>(undefined);
+export const MethodProvider = MethodContext.Provider;
+
+export interface MethodPropsWithType extends Omit<py.MethodDeclarationBaseProps, "name"> {
+  type: Operation;
+  name?: string;
+  doc?: Children;
+  parametersMode?: "prepend" | "append" | "replace";
+  methodType?: "method" | "class" | "static";
+  abstract?: boolean;
+}
+
+export type MethodProps = MethodPropsWithType | py.MethodDeclarationBaseProps;
+
+function createDocElement($: Typekit, props: MethodProps) {
```

Comment:
Shall we combine the two `createDocElement` functions? The only difference is `py.MethodDoc` vs. `py.ClassDoc`.